File: VB/RandomSurface/ViewModel/RandomDataViewModel.vb
#Region "#ViewModel"
Imports System
Imports System.Collections.ObjectModel

Namespace Bar3DChart

    Public Class RandomDataViewModel

        Public Property DataPoints As ObservableCollection(Of Point)

        Public Sub New()
            Dim barsCount As Integer = 200
            Dim maxValue As Integer = 200
            DataPoints = GenerateData(barsCount, maxValue)
        End Sub
    End Class

    Public Module DataGenerator

        Public Function GenerateData(ByVal pointCount As Integer, ByVal maxValue As Integer) As ObservableCollection(Of Point)
            Dim rand As Random = New Random()
            Dim bars As ObservableCollection(Of Point) = New ObservableCollection(Of Point)()
            For i As Integer = 0 To pointCount - 1
                Dim bar As Point
                Dim x As Double = rand.NextDouble() * maxValue * 2
                Dim y As Double = rand.NextDouble() * maxValue * 2
                Dim z As Double = rand.NextDouble() * maxValue
                bar = New Point(x, y, z)
                bars.Add(bar)
            Next

            Return bars
        End Function
    End Module
End Namespace
#End Region  ' #ViewModel

File: CS/RandomSurface/ViewModel/RandomDataViewModel.cs
#region #ViewModel
using System;
using System.Collections.ObjectModel;

namespace Bar3DChart {
    public class RandomDataViewModel {
        public ObservableCollection<Point> DataPoints { get; set; }
        public RandomDataViewModel() {
            int barsCount = 200;
            int maxValue = 200;
            this.DataPoints = DataGenerator.GenerateData(barsCount, maxValue);
        }
    }
    public static class DataGenerator {
        public static ObservableCollection<Point> GenerateData(int pointCount, int maxValue) {
            Random rand = new Random();
            ObservableCollection<Point> bars = new ObservableCollection<Point>();
            for (int i = 0; i < pointCount; i++) {
                Point bar;
                double x = rand.NextDouble() * maxValue * 2;
                double y = rand.NextDouble() * maxValue * 2;
                double z = rand.NextDouble() * maxValue;
                bar = new Point(x, y, z);
                bars.Add(bar);
            }
            return bars;
        }
    }
}
#endregion #ViewModel